File: tests/fixtures/debug_fail_target.py
"""Intentionally buggy script used to verify debug capture works."""


def process_order(items: list[str]) -> str:
    if not items:
        total = None
    else:
        total = len(items) * 10
    # BUG: total is None when items is empty, but we treat it as int
    return f"Total: {total + 5}"  # TypeError: unsupported operand type(s) for +: 'NoneType' and 'int'


def helper() -> None:
    # This function has a local variable that should appear in the stack
    multiplier = 2
    _ = multiplier


if __name__ == "__main__":
    helper()
    result = process_order([])
    print(result)

File: tests/fixtures/debug_fail_target.js
// Intentionally buggy JavaScript target for debug capture verification

function processOrder(items) {
    // BUG: items is null, accessing .total will throw
    const total = items.total;
    return total + 5;
}

function helper() {
    const multiplier = 2;
    return multiplier;
}

helper();  // Run helper to generate stack frames
const result = processOrder(null);  // Pass null to trigger TypeError
console.log(result);  // This won't be reached